src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosDbCollectionPhysicalPartitionInfo.cs

```diff
@@ -92,46 +98,18 @@ private async Task<int> GetPhysicalPartitionCount(CancellationToken cancellation
             using HttpClient client = _httpClientFactory.CreateClient();
 
             string host = _dataStoreConfiguration.Host;
-            string key = _dataStoreConfiguration.Key;
-
-            if (string.IsNullOrWhiteSpace(host))
-            {
-                if (string.IsNullOrWhiteSpace(key))
-                {
-                    host = CosmosDbLocalEmulator.Host;
-                    key = CosmosDbLocalEmulator.Key;
-                }
-                else
-                {
-                    Ensure.That(host, $"{nameof(CosmosDataStoreConfiguration)}.{nameof(CosmosDataStoreConfiguration.Host)}").IsNotNullOrEmpty();
-                }
-            }
-            else if (string.IsNullOrWhiteSpace(key))
-            {
-                Ensure.That(key, $"{nameof(CosmosDataStoreConfiguration)}.{nameof(CosmosDataStoreConfiguration.Key)}").IsNotNullOrEmpty();
-            }
```

Comment:
I think we should keep the same behavior but using the new **tokenProvider**.

We might need to add a new variable called "**token**" in [CosmosDBLocalEmuletor](https://github.com/microsoft/fhir-server/blob/fb9836bb14817f50977860217ff69a034493e90d/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosDbLocalEmulator.cs#L8) and make sure we clean up "**key**" in case it is not needed anymore. If "token" is not longer needed (because now you use _new DefaultAzureCredential()_), then can we clean up all key references?

I think we can keep the same structure because Ensure.That() will throw an exception in case host or token are null or white space, as now, it will just check **host** and not **token**.

Reply:
Do we still need "[key](https://github.com/microsoft/fhir-server/blob/fb9836bb14817f50977860217ff69a034493e90d/src/Microsoft.Health.Fhir.CosmosDb.Core/Configs/CosmosDataStoreConfiguration.cs#L15)"?

Reply:
@abiisnn We had removed "Key" reference from everywhere.
 Also, in CosmosDBLocalEmulator class, we do not need token property as we are using DefaultAzureCredentials, and it generates token internally.